tests/test_manager.rs

```diff
@@ -0,0 +1,82 @@
+#![feature(custom_derive, plugin)]
+#![plugin(serde_macros)]
+extern crate foxbox_taxonomy;
+extern crate foxbox_thinkerbell;
+extern crate serde;
+extern crate transformable_channels;
+
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use transformable_channels::mpsc::*;
+
+use foxbox_thinkerbell::compile::ExecutableDevEnv;
+use foxbox_thinkerbell::manager::*;
+
+use foxbox_taxonomy::api::{ API, Error as APIError, WatchEvent, TargetMap, ResultMap };
+use foxbox_taxonomy::selector::*;
+use foxbox_taxonomy::services::*;
+use foxbox_taxonomy::util::{ Exactly, Id };
+use foxbox_taxonomy::values::{ Range, Value };
+
+
+struct WatchGuard;
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct TestEnv;
+impl ExecutableDevEnv for TestEnv {
+    type API = ShimAPI;
+    type WatchGuard = WatchGuard;
+
+    fn api(&self) -> Self::API {
+        ShimAPI
+    }
+}
+
+struct ShimAPI;
+
+impl API for ShimAPI {
```

Comment:
Actually, as discussed with @JohanLorenzo, I finished this task. You can now use `fake_env::FakeEnv` as an environment to test compilation (and execution, but you probably don't need to).